api/v1alpha1/dnspolicy_types.go

```diff
@@ -0,0 +1,345 @@
+}
+
+type GeoCode string
+
+func (gc GeoCode) IsDefaultCode() bool {
+	return gc == DefaultGeo
+}
+
+func (gc GeoCode) IsWildcard() bool {
+	return gc == WildcardGeo
+}
+
+type LoadBalancingGeo struct {
+	// defaultGeo is the country/continent/region code to use when no other can be determined for a dns target cluster.
+	//
+	// The values accepted are determined by the target dns provider, please refer to the appropriate docs below.
+	//
+	// Route53: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-geo.html
+	// +required
+	DefaultGeo string `json:"defaultGeo,omitempty"`
+}
+
+// DNSPolicyStatus defines the observed state of DNSPolicy
+type DNSPolicyStatus struct {
+
+	// conditions are any conditions associated with the policy
+	//
+	// If configuring the policy fails, the "Failed" condition will be set with a
+	// reason and message describing the cause of the failure.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// observedGeneration is the most recently observed generation of the
+	// DNSPolicy.  When the DNSPolicy is updated, the controller updates the
+	// corresponding configuration. If an update fails, that failure is
+	// recorded in the status condition
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	HealthCheck *HealthCheckStatus `json:"healthCheck,omitempty"`
+}
+
+var _ common.KuadrantPolicy = &DNSPolicy{}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="gateway.networking.k8s.io/policy=direct"
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.conditions[0].reason`,description="DNSPolicy Status",priority=2
+// +kubebuilder:printcolumn:name="TargetRefKind",type="string",JSONPath=".spec.targetRef.kind",description="Type of the referenced Gateway API resource",priority=2
+// +kubebuilder:printcolumn:name="TargetRefName",type="string",JSONPath=".spec.targetRef.name",description="Name of the referenced Gateway API resource",priority=2
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+// DNSPolicy is the Schema for the dnspolicies API
+type DNSPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DNSPolicySpec   `json:"spec,omitempty"`
+	Status DNSPolicyStatus `json:"status,omitempty"`
+}
+
+func (p *DNSPolicy) GetWrappedNamespace() gatewayapiv1.Namespace {
+	return gatewayapiv1.Namespace(p.Namespace)
+}
+
+func (p *DNSPolicy) GetRulesHostnames() []string {
+	return make([]string, 0)
+}
+
+func (p *DNSPolicy) GetTargetRef() gatewayapiv1alpha2.PolicyTargetReference {
+	return p.Spec.TargetRef
+}
+
+func (p *DNSPolicy) Kind() string { return p.TypeMeta.Kind }
+
+// Validate ensures the resource is valid. Compatible with the validating interface
+// used by webhooks
+func (p *DNSPolicy) Validate() error {
+	if p.Spec.TargetRef.Group != "gateway.networking.k8s.io" {
```

Comment:
nit:
```suggestion
	if p.Spec.TargetRef.Group != gatewayapiv1.GroupName {
```

Reply:
There's actually a couple other cases in api/v1alpha1:
- https://github.com/Kuadrant/kuadrant-operator/blob/unified_kuadrant/api/v1alpha1/dnspolicy_types.go#L292
- https://github.com/Kuadrant/kuadrant-operator/blob/unified_kuadrant/api/v1alpha1/tlspolicy_types.go#L148
- https://github.com/Kuadrant/kuadrant-operator/blob/unified_kuadrant/api/v1alpha1/tlspolicy_types.go#L191

There are other cases in the codebase (not introduced by this change) but they are addressed by #396 I think